modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/display/context/HeaderFragmentDisplayContext.java

```diff
@@ -0,0 +1,509 @@
+				_httpServletRequest)
+		).setParameter(
+			"commerceOrderId", commerceOrder.getCommerceOrderId()
+		).buildString();
+	}
+
+	public long getCommerceAccountId() {
+		long accountEntryId = 0;
+
+		if (_accountEntry != null) {
+			accountEntryId = _accountEntry.getAccountEntryId();
+		}
+
+		return accountEntryId;
+	}
+
+	public CommerceOrder getCommerceOrder() throws PortalException {
+		long commerceOrderId = getCommerceOrderId();
+
+		if (commerceOrderId > 0) {
+			return _commerceOrderService.fetchCommerceOrder(
+				getCommerceOrderId());
+		}
+
+		return _commerceOrderService.fetchCommerceOrder(
+			ParamUtil.getString(_httpServletRequest, "commerceOrderUuid"),
+			_cpRequestHelper.getCommerceChannelGroupId());
+	}
+
+	public long getCommerceOrderId() {
+		return ParamUtil.getLongValues(
+			_httpServletRequest,
+			"_com_liferay_commerce_order_content_web_internal_portlet_" +
+				"CommerceOpenOrderContentPortlet_commerceOrderId")[0];
+	}
+
+	public List<HeaderActionModel> getHeaderActionModels() throws Exception {
+		List<HeaderActionModel> headerActionModels = new ArrayList<>();
+
+		CommerceOrder commerceOrder = getCommerceOrder();
+
+		if (isShowRetryPayment()) {
+			headerActionModels.add(
+				new HeaderActionModel(
+					"btn-primary", null, getRetryPaymentURL(), null,
+					"retry-payment"));
+		}
+
+		CommerceOrderStatus currentCommerceOrderStatus =
+			_commerceOrderEngine.getCurrentCommerceOrderStatus(commerceOrder);
+
+		if ((currentCommerceOrderStatus == null) ||
+			!currentCommerceOrderStatus.isComplete(commerceOrder) ||
+			(currentCommerceOrderStatus.getKey() ==
+				CommerceOrderConstants.ORDER_STATUS_CANCELLED) ||
+			(currentCommerceOrderStatus.getKey() ==
+				CommerceOrderConstants.ORDER_STATUS_IN_PROGRESS)) {
+
+			return headerActionModels;
+		}
+
+		if (!commerceOrder.isOpen()) {
+			headerActionModels.add(
+				new HeaderActionModel(
+					"btn-primary", null, getOrderDetailURL(), null, "reorder"));
+		}
+
+		List<CommerceOrderStatus> commerceOrderStatuses =
+			_commerceOrderEngine.getNextCommerceOrderStatuses(commerceOrder);
+
+		for (CommerceOrderStatus commerceOrderStatus : commerceOrderStatuses) {
+			if ((commerceOrderStatus.getKey() ==
+					CommerceOrderConstants.ORDER_STATUS_SHIPPED) ||
+				!commerceOrderStatus.isValidForOrder(commerceOrder) ||
+				!commerceOrderStatus.isTransitionCriteriaMet(commerceOrder)) {
+
+				continue;
+			}
+
+			String buttonCssClass = null;
+			String id = null;
+			String label;
+			String href = null;
+
+			if (commerceOrderStatus.getKey() ==
+					CommerceOrderConstants.ORDER_STATUS_IN_PROGRESS) {
+
+				if (!hasPermission(
+						CommerceOrderActionKeys.
+							CHECKOUT_OPEN_COMMERCE_ORDERS)) {
+
+					continue;
+				}
+
+				label = "checkout";
+				href = getCheckoutOrderURL();
+
+				if (!commerceOrder.isApproved()) {
+					label = "submit";
+
+					/*					href = PortletURLBuilder.create(
+											getTransitionOrderPortletURL(commerceOrder)
```

Comment:
commented out the implementation for now because of the `portletReponse` blocker.